persists/backend_server/server.bal

```diff
@@ -0,0 +1,65 @@
+import ballerina/http;
+import ballerina/persist;
+import ballerina/random;
+
+final Client orderDatabase = check new;
+
+@http:ServiceConfig {
+    cors: {
+        allowOrigins: ["*"]
+    }
+}
+service /sales on new http:Listener(9090) {
+    // Example: http://localhost:9090/sales/orders
+    isolated resource function get orders() returns Order[]|error {
+        return from Order entry in orderDatabase->/orders(targetType = Order)
+            select entry;
+    };
+
+    // Example: http://localhost:9090/sales/orders/HM-238
+    isolated resource function get orders/[string id]() returns Order|http:BadRequest {
+        Order|error orderEntry = orderDatabase->/orders/[id];
+        if orderEntry is Order {
+            return orderEntry;
+        }
+        return <http:BadRequest>{
+            body: {
+                message: string `Error while inserting the order, ${orderEntry.message()}`
+            }
+        };
+    };
+
+    // Example: http://localhost:9090/sales/cargos/HM-238/orders
+    isolated resource function get cargos/[string cargoId]/orders() returns Order[]|error {
+        return from Order entry in orderDatabase->/orders(targetType = Order)
+            where entry.cargoId == cargoId
+            where entry.cargoId == cargoId
+            order by entry.quantity descending
+            select entry;
+    };
+
+    // Example: http://localhost:9090/sales/orders
+    isolated resource function post orders(Order orderEntry) returns http:Ok|http:InternalServerError|http:BadRequest|error {
```

Comment:
Why are we sending both `http:InternalServerError` and `error` here? My suggestion is to stick with one here and use that pattern for all resource functions.

Reply:
updated